flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/PulsarWriter.java

```diff
@@ -243,5 +261,33 @@ public void flush(boolean endOfInput) throws IOException {
     public void close() throws Exception {
         // Close all the resources and throw the exception at last.
         closeAll(metadataListener, producerRegister);
+        closed = true;
```

Comment:
I'm not sure if this is the right way to signal shutdown to timerService. But I think it's ok to stop the timer Service even when closers are still functioning. Need to learn more about closers